source/elements/oneDPL/source/sycl_kernels_api.rst

```diff
@@ -46,17 +48,112 @@ Supported functionality:
    - ``cauchy_distribution``
    - ``extreme_value_distribution``
 
-Additionally, ``sycl::vec<>`` can be used as the result type for engines, engine adaptors, and distributions.
+``linear_congruential_engine`` and ``subtract_with_carry_engine`` satisfy the uniform random bit generator requirements.
 
 Limitations:
 ------------
+
 The following deviations from the `C++ Standard`_ may apply:
 
 - ``random_device`` and ``seed_seq`` classes and related APIs in other classes are not required;
 - ``operator>>()``, ``operator<<()``, ``operator==()`` are not required;
 - specifying the size of a random number engine's state is not required;
 - distributions are only required to operate with floating point types applicable to supported SYCL devices.
 
+Extensions:
+-----------
+
+As an extension to the `C++ Standard`_, ``sycl::vec<Type, N>`` can be used as the data type template parameter for
+engines, engine adaptors, and distributions, where ``Type`` is one of data types supported by the corresponding
+class template in the standard. For such template instantiations, the ``result_type`` is also defined to
+``sycl::vec<Type, N>``.
+
+Engines, engine adaptors, and distributions additionally define ``scalar_type``, equivalent to the following:
+
+- ``using scalar_type = typename result_type::element_type;`` if ``result_type`` is ``sycl::vec<Type, N>``,
+- otherwise, ``using scalar_type = result_type;``
+
+The ``scalar_type`` is used instead of ``result_type`` in all contexts where a scalar data type is expected, including
+
+- the type of configuration parameters and properties,
+- the seed value type,
+- the input parameters of constructors,
+- the return value type of ``min()`` and ``max()`` methods, etc.
+
+Since ``scalar_type`` is the same as ``result_type`` except for template instantiations with ``sycl::vec``,
+class templates still meet the applicable requirements of the `C++ Standard`_.
+
+When instantiated with ``sycl::vec<Type,N>``, ``linear_congruential_engine`` and ``subtract_with_carry_engine`` may not
+formally satisfy the uniform random bit generator requirements defined by the `C++ Standard`_. Instead, the following
+alternative requirements apply: for an engine object ``g`` of type ``G``,
+
+- ``G::scalar_type`` is an unsigned integral type same as ``sycl::vec<Type,N>::element_type``,
```

Comment:
```suggestion
alternative requirements apply. 

For an engine object ``g`` of type ``G``:

- ``G::scalar_type`` is an unsigned integral type same as ``sycl::vec<Type,N>::element_type``,
```

Reply:
I am afraid, in this way it's not quite clear what the alternative requirements are. Let's try improving it later.

Reply:
Postpone the change